### DISCUSSION TITLE
Do you have a road map of this library?

### DISCUSSION THREAD

**Discussion by ilyanikulin** [visgl/react-google-maps#100] [General]
Do you have a road map of this library?

**Comment by usefulthink** [comment]
Nothing formal yet, but here's the plan for the near future:

We aim to release the 1.0 version early next year. 

That release should mainly establish a set of core-APIs that should continue to work as long as there are no fundamental changes to the way React or the Google Maps API work (both very unlikely). 

This includes:

   - **core functionality:** everything around the `APIProvider`, the `Map` component, and its integration in a react-application (state handling and events, controlled and uncontrolled usage, ...). This also includes the hooks used to work with these components.

   - **extensible library:** everything that is needed to implement additional features, either in application code or as an additional library.

   - **tests, documentation and examples:** the documentation should be complete and consistent and be accompanied by examples for a variety of use-cases.

Everything beyond that is considered "nice to have" for now, we have to see how much additional features will added in time for the 1.0 release. As it is with features, those can be added at a later date and can be released with 1.x feature releases.

**Reply by jmdav1s** [reply]
Sorry to bump this one I know not much time has passed, any updates?

**Reply by usefulthink** [reply]
We just got back to working on this after the holiday break. Give it another month...

**Reply by lukas-siarny** [reply]
Hi there, could you please provide an update on when the first stable version will be released? We've switched from using the https://www.npmjs.com/package/@react-google-maps/api library to this new one. Everything is functioning well locally, but we're hesitant to go into production given its current alpha phase status.

**Reply by usefulthink** [reply]
Unfrotunately, I don't yet have a release date to share but I am relatively sure that there'll be no more major breaking changes to the core feature-set (i.e. APIProvider, Map and the related hooks). 

The events for Marker and AdvancedMarker could still receive a slight overhaul (i.e. adding support for AdvancedMarker DOM-events and maybe wrapping the events into something uniform like we did with the map-events). Then there are a couple of quality-of-life improvements I'm thinking about adding, but that wouldn't be breaking changes.

**Reply by jmdav1s** [reply]
We have pulled the trigger and used this in production on a relatively high traffic commercial site as well.  super keen to get to a stable version.  thank you so much for all your hard work!!
